components/ModEditHealthcareProfessionalSection.vue

```diff
@@ -238,52 +204,114 @@ const specialtyList = Object.values(Specialty) as Specialty[]
+        sectionFields.localizedMiddleName = ''
+        sectionFields.nameLocale = Locale.EnUs
     }
 }
 
-const handleRemoveHealthcareProfessionalName = (index: number) => {
-    professionalNameArray.splice(index, 1)
-}
+onBeforeMount(async () => {
+    isHealthcareProfessionalInitialized.value = false
+
+    /**
+    Set the variable to useToast when the before the component mounts
+    since vue-taostification is only available on the client.
+    If not done this way the build fails
+     */
+    toast = useToast()
+
+    // Wait for the route to be fully resolved
+    await nextTick()
+    // Ensure the route param `id` is available before proceeding
+    const id = route.params.id
+
+    if (!id) {
+        console.error(t('modHealthcareProfessionalSection.errorMessageHealthcareProfessionalId'))
+        toast.error(t('modHealthcareProfessionalSection.errorMessageHealthcareProfessionalId'))
+        return
+    }
+
+    if (!healthcareProfessionalsStore.healthcareProfessionalsData.length) {
+        await healthcareProfessionalsStore.getHealthcareProfessionals()
+    }
+
+    healthcareProfessionalsStore.selectedHealthcareProfessionalId = id as string
+
+    healthcareProfessionalsStore.setSelectedHealthcareProfessional(healthcareProfessionalsStore.selectedHealthcareProfessionalId)
+
+    healthcareProfessionalsStore
+        .initializeHealthcareProfessionalValues(healthcareProfessionalsStore.selectedHealthcareProfessionalData)
+
+    isHealthcareProfessionalInitialized.value = true
+
+    await nextTick()
+})
 
 onMounted(() => {
     multiSelectWithoutKeyboard(
         '#healthcare-professional-accepted-insurances',
-        ref(healthcareProfessionalSectionFields.healthcareProfessionalAcceptedInsurances),
+        healthcareProfessionalAcceptedInsurancesArray,
         selectedInsurance
     )
 
     multiSelectWithoutKeyboard(
         '#healthcare-professional-degrees',
-        ref(healthcareProfessionalSectionFields.healthcareProfessionalDegrees),
+        healthcareProfessionalDegreesArray,
         selectedDegree
     )
 
     multiSelectWithoutKeyboard(
         '#healthcare-professional-specialties',
-        ref(healthcareProfessionalSectionFields.healthcareProfessionalSpecialties),
+        healthcareProfessionalSpecialtiesArray,
         selectedSpecialty
     )
 
     multiSelectWithoutKeyboard(
         '#healthcare-professional-locales',
-        ref(healthcareProfessionalSectionFields.healthcareProfessionalLocales),
+        healthcareProfessionalSpokenLanguages,
         selectedLocale
     )
 })
+
+watch(
+    () => [
+        healthcareProfessionalAcceptedInsurancesArray.value,
+        healthcareProfessionalDegreesArray.value,
+        healthcareProfessionalSpecialtiesArray.value,
+        healthcareProfessionalSpokenLanguages.value
+    ] as [
+        Insurance[],
+        Degree[],
+        Specialty[],
+        Locale[]
+    ],
+    ([newInsurances, newDegrees, newSpecialties, newLocales]) => {
```

Comment:
might be good to explain why this watch is needed. (intuitively something would think they auto update if they're `ref`s)